src/Avalonia.Base/Input/PointerGesture.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Text;
+using Avalonia.VisualTree;
+
+namespace Avalonia.Input;
+
+public class PointerGesture : IEquatable<PointerGesture>
+{
+    public PointerGesture(MouseButton button, KeyModifiers modifiers = KeyModifiers.None)
+    {
+        Button = button;
+        KeyModifiers = modifiers;
+    }
+
+    public bool Equals(PointerGesture? other)
+    {
+        if (ReferenceEquals(null, other)) return false;
+        if (ReferenceEquals(this, other)) return true;
+
+        return Button == other.Button && KeyModifiers == other.KeyModifiers;
+    }
+
+    public override bool Equals(object? obj)
+    {
+        if (ReferenceEquals(null, obj)) return false;
+        if (ReferenceEquals(this, obj)) return true;
+
+        return obj is KeyGesture gesture && Equals(gesture);
+    }
+
+    public override int GetHashCode()
+    {
+        unchecked
+        {
+            return ((int)Button * 397) ^ (int)KeyModifiers;
+        }
+    }
+
+    public static bool operator ==(PointerGesture? left, PointerGesture? right)
+    {
+        return Equals(left, right);
+    }
+
+    public static bool operator !=(PointerGesture? left, PointerGesture? right)
+    {
+        return !Equals(left, right);
+    }
+
+    public MouseButton Button { get; }
+
+    public KeyModifiers KeyModifiers { get; }
+
+    public override string ToString()
+    {
+        var s = new StringBuilder();
+
+        static void Plus(StringBuilder s)
+        {
+            if (s.Length > 0)
+            {
+                s.Append("+");
+            }
+        }
+
+        if (KeyModifiers.HasAllFlags(KeyModifiers.Control))
+        {
+            s.Append("Ctrl");
```

Comment:
We recently accepted a new key formatter API that helps to avoid hardcoding these key strings. It should be integrated here as well https://github.com/AvaloniaUI/Avalonia/pull/15828

Reply:
Fixed - in latest commits I just reuse the code from KeyGesture.
But now I have some doubts about string format. Mouse buttons have rather short names, so
Ctrl+Left may look confusing, especially because there may be the same KeyGesture. Need to either remove it for now, or agree on some distinguishable button naming.